src/main/scala/com/softwaremill/mqperf/mq/ArtemisMq.scala

```diff
@@ -0,0 +1,18 @@
+package com.softwaremill.mqperf.mq
+
+import javax.jms.ConnectionFactory
+
+import com.softwaremill.mqperf.config.TestConfig
+import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory
+
+class ArtemisMq(val testConfig: TestConfig) extends Jms2Mq {
+
+  override lazy val connectionFactory: ConnectionFactory = {
+    val hosts = "failover:(" + testConfig.brokerHosts.map(h => s"tcp://$h:61616").mkString(",") + ")"
+    val cf = new ActiveMQConnectionFactory(hosts)
+    cf.setConfirmationWindowSize(10485760)
+    cf.setBlockOnDurableSend(false)
```

Comment:
Best contact for examples in the examples package would be clebert suconic, andrew taylor, martyn taylor.

The two sections that describe this is:
https://activemq.apache.org/artemis/docs/2.1.0/clusters.html
https://activemq.apache.org/artemis/docs/2.1.0/ha.html

here is a sample deployment diagram we have for once cluster we use in our org, we have 3 master and 3 slaves, we doe this for separation reasons.
https://gist.github.com/michaelandrepearce/170e09d8895cd165c7a956d573c5cbad

if you note we have the slaves and masters separated but it is possible to actually co-locate the slave and master so you can run just three brokers where each is a master and slave, using colocated replication.

```
<ha-policy>
   <replication>
      <colocated/>
   </replication>
</ha-policy>
```